### PR TITLE
Return highlighted excerpt of result (again)

### PR DESCRIPTION
**Pull request by ceritium**
Basically I adapted https://github.com/Casecommons/pg_search/pull/211

### CODE REVIEW

**Comment by ceritium** [comment]
Ping!

BTW: I don't know what I should do with rubocop and the errors in travis, on my machine works fine.

**Comment by amarshall** [comment]
The RuboCop failure was an issue on master which I’ve just fixed. If you rebase your changes and re-push your build should pass.

**Comment by ceritium** [comment]
Thanks @amarshall

**Comment by nertzy** [comment]
By the way I don't want to come across as harsh. I really appreciate the work you've put into this pull request. I think we can work through some small details and then get this in for sure.

File: lib/pg_search/features/tsearch.rb [hunk 1]
```diff
@@ -28,8 +24,47 @@ def rank
...
+        headline_options = map_headline_options
+        headline_options.map{|key, value| "#{key} = #{value}" if value }.compact.join(", ")
+      end
+
+      def map_headline_options
+        {
+          "StartSel" => options[:highlight][:start_sel],
+          "StopSel" => options[:highlight][:stop_sel],
+          "MaxFragments" => options[:highlight][:max_fragments]
+        }
```
Comment (ajb) [inline lib/pg_search/features/tsearch.rb]: What do you think of allowing other options? 

From [the docs](http://www.postgresql.org/docs/current/static/textsearch-controls.html#TEXTSEARCH-HEADLINE):

- FragmentDelimiter
- HighlightAll
- ShortWord
- MaxWords, MinWords

**Comment by ajb** [comment]
I haven't been active on this repo recently, but just chiming in to say that this looks solid, and it would be something that I'd use personally at @dobtco.

File: lib/pg_search/features/tsearch.rb (hunk 1 above)
Comment (ceritium) [inline lib/pg_search/features/tsearch.rb]: It is good idea, also I think that the options could be passed in the query and overwrite the default configuration.

**Comment by olimart** [comment]
:+1: nice. Thanks.

**Pull request merged**